src/cryptonote_basic/miner.cpp

```diff
@@ -575,7 +591,21 @@ namespace cryptonote
         continue; // if interrupted because stop called, loop should end ..
       }
 
-      bool on_ac_power = !on_battery_power();
+      boost::tribool battery_powered(on_battery_power());
+      bool on_ac_power = false;
+      if(indeterminate( battery_powered ))
+      {
+        // name could be better, only ignores battery requirement if we failed
+        // to get the status of the system
+        if( m_ignore_battery )
```

Comment:
The way I wanted the ignore_battery option to work was really more like "ignore if we can't figure out the battery status"